### PR TITLE
Delete button for simulations and other fixes

### PR DESCRIPTION
**Pull request by ThomasOliverKimble**
- Added delete moved refresh/sync buttons in _simulation_ tab with new modals
- Fixed pagination bug on refresh
- Icons changed for _Demo_ in _simulation_ tab
- Slight code restructure in `webots-cloud.js`

### CODE REVIEW

File: httpdocs/css/webots-cloud.css [hunk 1]
```diff
@@ -6,11 +6,11 @@ body {
   height: 100%;
 }
 body:after{
-  content: "beta version";
+  content: "testing";
```
Comment (BenjaminDeleze) [inline httpdocs/css/webots-cloud.css]:
```suggestion
  content: "beta version";
```

As you will merge it in beta

File: httpdocs/css/webots-cloud.css [hunk 2]
```diff
@@ -6,11 +6,11 @@ body {
...
   position: fixed;
   width: 260px;
   height: 25px;
-  background: #e00f00;
+  background: purple;
```
Comment (BenjaminDeleze) [inline httpdocs/css/webots-cloud.css]:
```suggestion
  background: #e00f00;
```
To go back to the beta layout

File: httpdocs/js/webots-cloud.js [hunk 3]
```diff
@@ -591,34 +552,127 @@ document.addEventListener('DOMContentLoaded', function() {
...
+              else if (data.status == 1)
+                project.load(`/simulation${(page > 1) ? ('?p=' + page) : ''}`);
+          });
+      });
+    }
+  }
+
   function deleteAnimation(event, type, project, page) {
-    const that = this;
+    console.log(JSON.stringify(project));
```
Comment (BenjaminDeleze) [inline httpdocs/js/webots-cloud.js]:
```suggestion
```

Is it a debug line that we can remove?

**Review by BenjaminDeleze — CHANGES_REQUESTED** [review]
Just some minors remarks but otherwise it looks good to me.

File: httpdocs/js/webots-cloud.js (hunk 3 above)
Comment (ThomasOliverKimble) [reply httpdocs/js/webots-cloud.js]: yes

**Comment by BenjaminDeleze** [comment]
Seems all good to me, but it would be nice if maybe @omichel have a quick look at it has he is more familiar than me with this code

File: httpdocs/ajax/project/delete.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+$id = intval($data->id);
+require '../../../php/database.php';
+$mysqli = new mysqli($database_host, $database_username, $database_password, $database_name);
+if ($mysqli->connect_errno)
+  error("Can't connect to MySQL database: $mysqli->connect_error");
+$mysqli->set_charset('utf8');
+$query = "DELETE FROM project WHERE id=$id";
+$mysqli->query($query) or error($mysqli->error);
+if ($mysqli->affected_rows === 0)
+  error('Could not delete animation');
```
Comment (omichel) [inline httpdocs/ajax/project/delete.php]:
```suggestion
  error('Could not delete simulation');
```

File: httpdocs/ajax/project/delete.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+// Until here all is good in the hood
+require '../../../php/simulation.php';
+die("{\"status\":1}");
+?>
```
Comment (omichel) [inline httpdocs/ajax/project/delete.php]:
```suggestion
?>

```

File: httpdocs/css/webots-cloud.css [hunk 6]
```diff
@@ -86,3 +86,16 @@ nav.pagination > a[disabled] {
...
+  color: lightgrey;
+}
+
+.synchronizable-icon:hover {
+  color: #444;
+}
+
+.fa-spin {
+  color: #444;
+}
```
Comment (omichel) [inline httpdocs/css/webots-cloud.css]:
```suggestion
}

```

File: httpdocs/js/webots-cloud.js [hunk 7]
```diff
@@ -75,21 +75,35 @@ document.addEventListener('DOMContentLoaded', function() {
...
     const page_limit = 10;
     if (active_tab === '')
       active_tab = 'animation';
 
+    mainContainer(project, active_tab);
+    initTabs();
+
+    project.content.querySelector('#add-a-new-scene').addEventListener('click', function(event) {addAnimation('S');});
+    project.content.querySelector('#add-a-new-animation').addEventListener('click', function(event) {addAnimation('A');});
+    project.content.querySelector('#add-a-new-project').addEventListener('click', function(event) {addSimulation();});
```
Comment (omichel) [inline httpdocs/js/webots-cloud.js:90]:
```suggestion
    project.content.querySelector('#add-a-new-scene').addEventListener('click', function(event) { addAnimation('S'); });
    project.content.querySelector('#add-a-new-animation').addEventListener('click', function(event) { addAnimation('A'); });
    project.content.querySelector('#add-a-new-project').addEventListener('click', function(event) { addSimulation(); });
```

File: httpdocs/js/webots-cloud.js [hunk 8]
```diff
@@ -164,29 +178,25 @@ document.addEventListener('DOMContentLoaded', function() {
...
-      const title = data.title === '' ? '<i>anonymous</i>' : data.title;
       let type;
-      let competitors;
       if (data.type == 'demo') {
-        type = `<i class="fas fa-grin-stars fa-lg" title="${data.type}"></i>`;
-        competitors = '';
+        type = `<i class="fas fa-chalkboard-teacher fa-lg" title="${data.type}"></i>`;
       } else {
-        const icon = (data.type == 'benchmark') ? 'award' : 'trophy';
+        const icon = (data.type == 'benchmark') ? 'chart-line' : 'question';
```
Comment (omichel) [inline httpdocs/js/webots-cloud.js]: I like the char-line icon for benchmarks, but I would like to keep the trophy for competitions, as if it is not a benchmark, it is a competition.
```suggestion
        const icon = (data.type == 'benchmark') ? 'chart-line' : 'trophy';  // if not a benchmark, it is a competition
```

**Review by omichel — CHANGES_REQUESTED** [review]
I have some doubts regarding the delete.php ajax request which might allow anyone to wipe out the simulation database very easily...

**Pull request closed without merging**